synapse/storage/databases/main/roommember.py

```diff
@@ -1636,6 +1697,84 @@ def __init__(
             columns=["user_id", "room_id"],
         )
 
+        self.db_pool.updates.register_background_update_handler(
+            "populate_participant_bg_update", self._populate_participant
+        )
+
+    async def _populate_participant(self, progress: JsonDict, batch_size: int) -> int:
+        """
+        Background update to populate column `participant` on `room_memberships` table
+
+         A 'participant' is someone who is currently joined to a room and has sent at least
+        one `m.room.message` or `m.room.encrypted` event.
```

Comment:
Pedantic, but may save us a headache down the line, as we love to use existing columns in weird and wonderful ways.

```suggestion
        one `m.room.message` or `m.room.encrypted` event.
        
        This background update will set the `participant` column across all rows in
        `room_memberships` based on the user's *current* join status, and if
        they've *ever* sent a message or encrypted event. Therefore one should
        never assume the `participant` column's value is based solely on whether
        the user participated in a previous "session" (where a "session" is defined
        as a period between the user joining and leaving). See
        https://github.com/element-hq/synapse/pull/18068#discussion_r1931070291
        for further detail.
```